lua/wire/flir.lua

```diff
@@ -37,126 +56,160 @@ if CLIENT then
 		["$pp_colour_brightness"] = 0
 	}
 
-	local function SetFLIRMat(ent)
-		if not IsValid(ent) then return end
+	--add and remove entities from the FLIR rendering stack
 
-		if (ent:GetMoveType() == MOVETYPE_VPHYSICS or ent:IsPlayer() or ent:IsNPC() or ent:IsRagdoll() or ent:GetClass() == "gmod_wire_hologram") and ent:GetColor().a > 0 then
-			ent.RenderOverride = FLIR.Render
-			FLIR.RenderStack[ent] = true
-		end
+	local function RemoveFLIR(ent)
+		FLIR.RenderStack[ent] = nil
+		if IsValid(ent) then ent.RenderOverride = nil end
```

Comment:
ent:IsValid()

Reply:
checking the wiki, I don't see a difference since isvalid checks ent:IsValid(). Is there a performance difference?